File: TypeScript/FrenchAddressBuilder.ts
import { AddressBuilder } from './AddressBuilder';
import { AddressData } from './AddressData';

export class FrenchAddressBuilder extends AddressBuilder{

  buildLine3() : void {
    this.address.addLine(this.addressData.zip + " " + this.addressData.city + "\n");
  }

    constructor(addressData: AddressData) {
    super(addressData);
    this.title_f = "Madame";
    this.title_m = "Monsieur";
  }
}

File: Java/FrenchAddressBuilder.java
public class FrenchAddressBuilder extends AddressBuilder {

    public FrenchAddressBuilder(AddressData addressData) {
        super(addressData);
        titleF = "Madame";
        titleM = "Monsieur";
    }

    @Override
    public void buildLine3(){
        address.addLine(addressData.zip() + " " + addressData.city() + "\n");
    }
}
